src/GraphQLParser/Visitors/SDLPrinter.cs

```diff
@@ -1017,13 +1022,25 @@ private static async ValueTask WriteEncodedStringAsync(TContext context, ROM val
                     await context.WriteAsync("\\u" + ((int)code).ToString("X4")).ConfigureAwait(false);
             }
             else if (code == '\\')
+            {
+                if (startIndexOfNotEncodedString != i)
+                    await context.WriteAsync(value.Slice(startIndexOfNotEncodedString, i - startIndexOfNotEncodedString)).ConfigureAwait(false);
+                startIndexOfNotEncodedString = i + 1;
+
                 await context.WriteAsync("\\\\").ConfigureAwait(false);
+            }
             else if (code == '"')
+            {
+                if (startIndexOfNotEncodedString != i)
+                    await context.WriteAsync(value.Slice(startIndexOfNotEncodedString, i - startIndexOfNotEncodedString)).ConfigureAwait(false);
+                startIndexOfNotEncodedString = i + 1;
+
                 await context.WriteAsync("\\\"").ConfigureAwait(false);
-            else
-                await context.WriteAsync(value.Slice(i, 1)/*code*/).ConfigureAwait(false); // TODO: no method for char
```

Comment:
Hot path: await + one-symbol slicing 

@Shane32 We should use the same approach for comments/block strings

Reply:
So now in case of "normal" strings this loop does nothing just incrementing counter - single write is done at the end of method:
`
 await context.WriteAsync(value.Slice(startIndexOfNotEncodedString, value.Span.Length - startIndexOfNotEncodedString)).ConfigureAwait(false);
`